src/rendering/renderers/shared/system/AbstractRenderer.ts

```diff
@@ -316,4 +318,29 @@ export class AbstractRenderer<PIPES, OPTIONS>
         writeable.renderPipes = null;
         writeable.runners = null;
     }
+
+    /**
+     * @deprecated since 8.0.0
+     * @param container - the container to render
+     * @param options - options to use when generating the texture
+     * @returns a texture
+     */
+    public generateTexture(container: Container, options?: IGenerateTextureOptions): Texture
+    {
+        deprecation(
+            v8_0_0,
+            'renderer.generateTexture() is deprecated, please use renderer.textureGenerator.generateTexture() directly'
+        );
+
+        return this._generateTexture(container, options);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    protected _generateTexture(container: Container, options?: IGenerateTextureOptions): Texture
+    {
+        // subclasses need to handle as the abstract renderer does not know the systems available
+        throw new Error('Not implemented');
+    }
```

Comment:
This can be simplified to 

```ts
public textureGenerator: GenerateTextureSystem;

generateTexture(container: Container, options?: IGenerateTextureOptions): Texture
{
    return this.textureGenerator.generateTexture({
        container,
        region: options?.region,
    });
}
```

We do this with `view: ViewSystem` as well so that `AbstractRenderer` has access to it and we wont need each renderer to implement the same thing

Reply:
👍